sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala

```diff
@@ -126,9 +126,12 @@ case class CoalesceShufflePartitions(session: SparkSession) extends AQEShuffleRe
   // data sources may request a particular advisory partition size for the final write stage
   // if it happens, the advisory partition size will be set in ShuffleQueryStageExec
   // only one shuffle stage is expected in such cases
-  private def advisoryPartitionSize(shuffleStages: Seq[ShuffleStageInfo]): Long = {
+  private def advisoryPartitionSize(coalesceGroup: CoalesceGroup): Long = {
+    if (coalesceGroup.hasExplodingJoin) {
+      return conf.getConf(SQLConf.COALESCE_PARTITIONS_MIN_PARTITION_SIZE)
```

Comment:
shall we add a new config like `explodingStageAdvisorySizeFactor` to make the advisory size easy to change ?

Reply:
isn't it harder to tune two configs (advisory size + advisory size factor for exploding) than just one advisory size for exploding? I didn't add a new config as I think `COALESCE_PARTITIONS_MIN_PARTITION_SIZE` has a reasonable default value to turn off coalesce for this case, and we do need to respect it when coalescing partitions.

Reply:
I thought if the query contain multi-stages and there are more than one stage contain exploding operators. People may want to change this config more fine-grained. That's why I tried to support plan fragment config before.

I'm fine to use the `COALESCE_PARTITIONS_MIN_PARTITION_SIZE`.

Reply:
Then we probably need to add reduce ratio which needs more time to test and build it correctly: https://github.com/apache/spark/pull/45357#pullrequestreview-1911078774

By having the reduce ratio, we can do coalescing differently for different coalesce groups. Maybe we can do it as a follow-up.

Reply:
Maybe we can respect `spark.sql.adaptive.coalescePartitions.parallelismFirst here`?

Reply:
`parallelismFirst` affects the `minNumPartitions`, seems to be a different thing here.